spec/gateways/email_gateway_spec.rb

```diff
@@ -1,14 +1,16 @@
 require 'support/notifications_service'
 
 describe EmailGateway do
+  subject(:email_gateway) { described_class.new }
+
+  let(:notification) { instance_spy(Notifications::Client, send_email: nil) }
+
   before do
     ENV['NOTIFY_API_KEY'] = 'dummy_key-00000000-0000-0000-0000-000000000000-00000000-0000-0000-0000-000000000000'
-
-    expect_any_instance_of(Notifications::Client).to \
-      receive(:send_email).with(notifications_payload)
+    allow(Notifications::Client).to receive(:new).and_return(notification)
   end
 
-  context 'confirmation emails' do
+  context 'with confirmation emails' do
```

Comment:
This isn't very clear to me, it says "EmailGateway with confirmation emails calls notifications" when you put it all together.

After reading the spec it seems something better could be "EmailGateway when sending confirmation emails calls the Notify client"